lib/facter/powershell_version.rb

```diff
@@ -3,9 +3,7 @@
 
 Facter.add(:powershell_version) do
   setcode do
-    if Puppet::Util::Platform.windows?
-      version = PuppetX::PuppetLabs::Dsc::PowerShellVersion.version
-      version
-    end
+    platform = Puppet::Util::Platform
+    platform.windows? ? PuppetX::PuppetLabs::Dsc::PowerShellVersion.version : 0
```

Comment:
Seems fine to me.

Note that `Puppet::Util::Platform` is a little weird, because it's not a list like `POSIX`, `SLES`, `AIX`, etc... but rather just a couple of seemingly random things - `windows?`, `default_paths`, `fips_enabled?` and `jruby`

https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/platform.rb#L7-L15